z-schema/index.d.ts

```diff
@@ -90,7 +132,19 @@ declare namespace ZSchema {
          */
         validate(json: any, schema: any, callback: (err: any, valid: boolean) => void): void;
 
+        /**
+         * Returns an Error object for the most recent failed validation, or null if the validation was successful.
+         */
         getLastError(): SchemaError;
-        getLastErrors(): SchemaError[];
+
+        /**
+         * Returns the error details for the most recent validation, or undefined if the validation was successful.
+         * This is the same list as the SchemaError.details property.
+         */
+        getLastErrors(): SchemaErrorDetail[];
     }
-}
\ No newline at end of file
+}
+
+declare module "z-schema" {
```

Comment:
@mhegazy  Could you give an example?  I originally tried this style:

```typescript
declare namespace ZSchema {
    export interface SchemaError extends Error {
        . . .
    }
    . . .
    export class Validator {
       . . .
    }
}
export = ZSchema.Validator;
```

...but I could not get it to work.  For example:

```typescript
import Validator = require('z-schema');
var validator: Validator = new Validator(options);
var error: ZSchema.SchemaError = validator.getLastError(); // error: Cannot find namespace ZSchema
```

Since the "import" statement brings in the "=" export, it seems like I would need some other expression to import the namespace, something like this:

```typescript
import Validator = require('z-schema');
import { ZSchema } from 'z-schema';
```

...but it doesn't work.  The ambient definition is the only syntax I could find that allows me to access the ZSchema namespace definitions.

I'd greatly appreciate any help/suggestions for a better solution.